java/src/jmri/jmrix/loconet/SlotManager.java

```diff
@@ -998,14 +1006,31 @@ public void writeCV(String cvNum, int val, jmri.ProgListener p) throws jmri.Prog
             } else if (getMode().equals(ProgrammingMode.REGISTERMODE)
                     || getMode().equals(ProgrammingMode.ADDRESSMODE)) {
                 pcmd = pcmd | 0x10;
+            } else if (getMode().equals(ProgrammingMode.DIRECTBITMODE)) {
+                pcmd = pcmd | 0x08;
             } else {
                 throw new jmri.ProgrammerException("mode not supported"); // NOI18N
             }
 
-            doWrite(CV, val, p, pcmd);   
+            doWrite(Integer.parseInt(cvNum), val, p, pcmd);
+
         }
     }
 
+    /**
+     * Write a CV via the Service Mode programmer.
+     *
+     * @param CV CV number
+     * @param val value to write to the CV
+     * @param p programmer
+     * @throws jmri.ProgrammerException if an unsupported programming mode is exercised
+     */
+    @Deprecated // 4.1.1
+    public void writeCV(int CV, int val, jmri.ProgListener p) throws jmri.ProgrammerException {
```

Comment:
This seems to have lost the changes in PR #6376.  That removed this method and merged it into the call just above.